packages/rps/src/redux/metamask/saga.ts

```diff
@@ -1,47 +1,52 @@
-import {cps, delay, put} from 'redux-saga/effects';
+import {call, put} from 'redux-saga/effects';
 import * as metamaskActions from './actions';
 import {MetamaskErrorType} from './actions';
 
-export default function* checkMetamask() {
-  if (typeof window.web3 !== 'object' || window.web3 === null) {
-    yield put(
-      metamaskActions.metamaskErrorOccurred({
-        errorType: MetamaskErrorType.NoWeb3,
-      })
-    );
-    return false;
-  }
-
-  const networks = {
-    development: {
-      networkId: '*', // match any network
-    },
-    main: {
-      networkId: 1,
-    },
-    ropsten: {
-      networkId: 3,
-    },
-    rinkeby: {
-      networkId: 4,
-    },
-    kovan: {
-      networkId: 42,
-    },
-  };
+interface NetworkNameToId {
+  [name: string]: {networkId: number};
+}
+const networks: NetworkNameToId = {
+  main: {
+    networkId: 1,
+  },
+  ropsten: {
+    networkId: 3,
+  },
+  rinkeby: {
+    networkId: 4,
+  },
+  kovan: {
+    networkId: 42,
+  },
+};
 
-  try {
+export default function* checkMetamask() {
+  if (typeof window.ethereum !== 'undefined') {
+    try {
+      window.ethereum.autoRefreshOnNetworkChange = false;
```

Comment:
Why false?

Reply:
This is just to clear the error from the console, since the setting will soon vanish (see link in commit message). We are now essentially turning off a feature that is soon to disappear, before rolling our own solution just below (since we rely on the reloading behaviour).